gap/perm.gi

```diff
@@ -159,6 +161,10 @@ FindHomMethodsPerm.BalTreeForBlocks := function(ri,G)
   upperhalf := blocks{[cut+1..nrblocks]};
   o := Concatenation(upperhalf);
   hom := ActionHomomorphism(G,o);
+
+  # Make a set so checking validatehomominput is faster
+  seto := Immutable(Set(o));
```

Comment:
Actually, often computing the `ActionHomomorphism` is also faster if one gives it a set. So perhaps just redefine `o` above?

Reply:
Actually, I did that and it broke other things, which assumed that `o` was still `Concatenation(upperhalf)`. I could add a comment to say why that wasn't changed.

Reply:
Ah, I see. Well, such a comment would then indeed be helpful :-)